djconnectwise/sync.py

```diff
@@ -217,11 +217,25 @@ def persist_page(self, records, results):
                     results.updated_count += 1
                 else:
                     results.skipped_count += 1
+
+                # Only add to synced_ids if:
+                # 1. This is not a ticket, OR
+                # 2. It's a project ticket/issue with an open project
+                # This prevents closed project tickets from being considered "synced"
+                # which we delete in prune_stale_records.
+                if (
+                    self.model_class is not models.TicketTracker
```

Comment:
Definitely remove this, persist page needs no updates to get this to work. Especially not any update that relies on a model class type. Raising an InvalidObjectException should be enough to block it from being added to synced_ids